.github/workflows/build-and-publish.yml

```diff
@@ -0,0 +1,92 @@
+name: Docker Image CI
+
+on:
+  push:
+    tags:
+     - v**
+  pull_request:
+    branches: [ master ]
+     
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: write
+      packages: write
+      
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v3
+      with:
+        submodules: recursive
```

Comment:
We shouldn't strictly need this, but I guess it doesn't hurt.

Since we got the transwarp dependency listed in `requirements.txt` now, the submodule has become obsolete. I think we'll remove it over time.

Reply:
I'm going to leave it for now, we an nuke it when the submodule goes away, or just let it stick around in case we need submodules in the future. As you said, it doesn't hurt anything